gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java

```diff
@@ -245,7 +262,27 @@ protected void postConnectionHook(
         // Saving history at gateway.
         queryHistoryManager.submitQueryDetail(queryDetail);
       } else {
-        log.debug("SKIPPING For {}", requestPath);
+        // Reading next uri response
+        if (requestPath.endsWith("/1") && requestPath.startsWith(V1_STATEMENT_PATH)) {
+          // This is first page reading. If response has StartTransaction Headers, save backend
+          // mapping with transaction id.
+          String startTransactionHeader = response.getHeader(STARTED_TRANSACTION_ID);
```

Comment:
Would be good to understand why transaction not started on the first call. This seems a little strange.

Reply:
I have asked this question in presto-slack - https://prestosql.slack.com/archives/CFLB9AMBN/p1561663907382700?thread_ts=1561483585.338800&cid=CFLB9AMBN 
I will explore the presto-code to see if I can find more info.